pkgs/by-name/be/beanhub-import/package.nix

```
@@ -1,36 +1,28 @@
 {
   lib,
   fetchFromGitHub,
-  buildPythonPackage,
-  pythonOlder,
-  pytestCheckHook,
-  beancount-black,
-  beancount-parser,
-  beanhub-extract,
-  jinja2,
-  poetry-core,
-  pydantic,
-  pytz,
-  pyyaml,
+  python3,
 }:
 
-buildPythonPackage rec {
+python3.pkgs.buildPythonPackage rec {
```

If you want to move this out form python3Packages and into by-name, this this is required:
```suggestion
python3.pkgs.buildPythonApplication rec {
```

---

I thought `buildPythonPackage` and `python3.pkgs.buildPythonPackage` are the same thing? Are they different?

And as this is a library, isn't the `buildPythonApplication` one for application, like command line tools?

---

When you move the package out of `pkgs/development/python-modules` into `pkgs/by-name`, you change the attribute name from from `python3Packages.beanhub-import` to `beanhub-import`, this means that if you want to use a non-standard python version (i.e. `python313` instead of just `python3`) you'll get conflicts. As such we don't put libraries in `by-name`, only applications. I assumed this must be some application and not a library due to the move.

---

> Are they different?
The derivation/resulting store path is the same, but some extra attributes will differ which determine how it gets added to your environment.

---

I see, I was thinking about getting the benefit of auto update, that's the major reason why I move it to `by-name` folder. I thought the nixpkg community is moving toward `by-name` style package, but didn't realize it doesn't cover the python library usecase. Thanks for letting me know. I will change it back shortly.

---

okay, I revert it back to the old format. only update version and use hatching instead of poetry.